### PR TITLE
Create User List UI functionality

### LINKED ISSUE #137: Implement Create user List UI functionality
Connect the create list ui to the backend
- [x] Create user list graphql mutation also takes in a rating system id
    - [x] Default rating system ids to indicate default rating systems such as 10 point discrete, 10 point continuous, 5 star rating, etc. 
- [x] When clicking `Create User List` from the create user list ui, create the list and take the user to the list page
- [x] The rating system drop down lists default rating systems

### PR DESCRIPTION
**Pull request by jvoucsb**
## Notes
- Created Default rating systems: 10 point decimal, 100 point decimal, 10 point discrete, Five stars, three smiley emojis
- Create List UI now works
- Lists the user's rating systems in dropdown menu along with the default rating menus

## Tests
- Unit tests
- Manual test

createlist

Closes #137

### CODE REVIEW

File: client/src/components/list/CreateUserList.tsx [hunk 1]
```diff
@@ -1,78 +1,104 @@
...
 
 
 const CreateUserList: React.FC<{}> = () => {
+  const [createUserList] = useCreateUserListMutation();
+  const { data: meData } = useMeQuery();
+  const toast = useToast();
+  const router = useRouter();
 
-    const { isOpen, onOpen, onClose } = useDisclosure()
+  const validationSchema = Yup.object({
```
Comment (rowantran) [inline client/src/components/list/CreateUserList.tsx:33]: where are the errors displayed if a field is not validated?

File: server/src/main/java/com/github/animelist/animelist/model/ratingsystem/RatingSystem.java [hunk 2]
```diff
@@ -49,7 +49,6 @@ public RatingSystem(String id, String name, ObjectId ownerId, Integer size, List
         var weightSum = subRatings.stream().mapToDouble(SubRating::getWeight).sum();
         var difference = WEIGHT_SUM_EXPECTED - weightSum;
         Assert.isTrue(difference >= 0 && difference <= WEIGHT_SUM_EPSILON, "subRatings weight sum must sum up to 1");
-        Assert.isTrue(name.matches("^[a-zA-Z0-9]+(?:[\\w -]*[a-zA-Z0-9]+)*$"),"Name must be AlphaNumeric");
```
Comment (rowantran) [inline server/src/main/java/com/github/animelist/animelist/model/ratingsystem/RatingSystem.java:52]: why did we remove this constraint?

**Review by rowantran — APPROVED** [review]
LGTM, left some questions about the code

File: client/src/components/list/CreateUserList.tsx (hunk 1 above)
Comment (rowantran) [reply client/src/components/list/CreateUserList.tsx:33]: I see in https://formik.org/docs/guides/validation that you can pass a function accepting `errors` when you use the Formik component, but not seeing how we display errors here

**Review by rowantran — APPROVED** [review]
questions answered in person, LGTM! excellent work

**Review by AHuangHe — APPROVED** [review]
LGTM!

**Pull request merged**